src/components/replay.js

```diff
@@ -0,0 +1,24 @@
+AFRAME.registerComponent("replay", {
+  init: function() {
+    this.playhead = 0;
+    this.poseIndex = 0;
+  },
+
+  tick: function(t, dt) {
+    let overflow = false;
+    while (!overflow && this.playhead >= this.poses[this.poseIndex].timestamp) {
```

Comment:
This seems odd, it appears to be setting the position/rotation/scale of the same element multiple times in a single frame. Seems like it should find the pose for the current playhead position and then set the position/rotation/scale once.

Reply:
Yes that seems more correct

Reply:
But did you consider that writing that version will require _slightly_ more work 😜